include/Ark/Compiler/MacroExecutors/ListExecutor.hpp

```diff
@@ -0,0 +1,34 @@
+/**
+ * @file ListExecutor.hpp
+ * @author Ray John Alovera (rakista112@gmail.com)
+ * @brief Executor for List Macros
+ * @version 0.1
+ * @date 2021-05-04
+ * 
+ * @copyright Copyright (c) 2021
+ * 
+ */
+
+#ifndef ARK_COMPILER_LISTEXECUTOR_HPP
+#define ARK_COMPILER_LISTEXECUTOR_HPP
+#include <Ark/Compiler/MacroExecutor.hpp>
+#include <unordered_map>
+#include <vector>
+#include <Ark/Compiler/Node.hpp>
+
+namespace Ark::internal
+{
+    /**
+     * @brief Handles List macros
+     * 
+    */
+    class ListExecutor : public MacroExecutor
+    {
+    public:
+        void execute(Node &node);
+        bool canHandle(Node &node);
```

Comment:
both should be marked override to specifiy that the methods are overriden and not just shadowing the parent method (even though it works without it, that will make it clearer to other devs that it's a virtual method implementation)